tensorflow_addons/seq2seq/decoder.py

```diff
@@ -253,16 +257,16 @@ def tracks_own_finished(self):
 
 
 def dynamic_decode(
-    decoder,
-    output_time_major=False,
-    impute_finished=False,
-    maximum_iterations=None,
-    parallel_iterations=32,
-    swap_memory=False,
-    training=None,
-    scope=None,
+    decoder: Decoder,
+    output_time_major: bool = False,
+    impute_finished: bool = False,
+    maximum_iterations: Optional[Number] = None,
+    parallel_iterations: int = 32,
+    swap_memory: bool = False,
+    training: Optional[bool] = None,
+    scope: Optional[str] = None,
     **kwargs
-):
+) -> tf.Tensor:
```

Comment:
I don't think we can easily express the return value as the state structure is arbitrary.

Reply:
Thanks @guillaumekln. In this case, may I have your suggestions?

Reply:
I would suggest to not type the return value.

Reply:
I think it's safe to put `Tuple[Any, Any, Any]` we can refine later on if necessary.

Reply:
Sure, thanks @gabrieldemarmiesse.